### ISSUE TITLE
-w still prompts for `Perform changes ? (y/n)` but changes were already written

### ISSUE THREAD

**Issue by xftroxgpx** [dosfstools/dosfstools#45]
```bash
sudo fsck.vfat -v -f -V -w -- /dev/sdb1
```
This still prompts for `Perform changes ? (y/n)` and answering `n` is useless because at least the second FAT(what I tested this on: invalidated 2nd FAT) was already written. A second run of the same command detects nothing wrong, but this time no prompt and thus `fs_flush(void)` is reached.

I'm on Arch Linux, dosfstools 4.1-1

Replacing '-w' with '-r' still prompts, as expected, and changes(second FAT at least) aren't written, as expected.

**Comment by xftroxgpx** [comment]
I suppose this would work(tested once) as a workaround:

```diff
diff -upr '--exclude=*.Po' dosfstools-4.1/src/fsck.fat.c dosfstools-4.1.mod/src/fsck.fat.c
--- dosfstools-4.1/src/fsck.fat.c	2016-05-10 11:58:39.000000000 +0200
+++ dosfstools-4.1.mod/src/fsck.fat.c	2017-05-05 14:42:02.572230002 +0200
@@ -207,7 +207,7 @@ int main(int argc, char **argv)
 exit:
     if (fs_changed()) {
 	if (rw) {
-	    if (interactive)
+	    if (interactive && !write_immed)
 		rw = get_key("yn", "Perform changes ? (y/n)") == 'y';
 	    else
 		printf("Performing changes.\n");
```

**Comment by andreasbombe** [comment]
Thanks, but I made some bigger changes to the interactive interface and already fixed this in 141b00998d.

**Comment by xftroxgpx** [comment]
Cool! I should switch to using the git version. Thanks!
